rust-wasm-id-allocator/distributed-id-allocator/src/compressor/persistence.rs

```diff
@@ -1,17 +1,25 @@
+use crate::id_types::SessionId;
+
 use super::IdCompressor;
 use postcard::from_bytes;
 use serde::{Deserialize, Serialize};
 
 pub(super) const DEFAULT_CLUSTER_CAPACITY: u64 = 512;
 
-pub(crate) fn deserialize(bytes: &[u8]) -> Result<IdCompressor, DeserializationError> {
+pub(crate) fn deserialize<FMakeSession>(
+    bytes: &[u8],
+    make_session_id: FMakeSession,
```

Comment:
Went with this pattern instead of passing in a SessionId itself as we only sometimes use it (when no local state).